src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -40,4 +56,28 @@ public void periodic() {
     public void run(double speed) {
         intakeMotor.set(speed);
     }
+
+    public OutColor read () {
+        Color reading = colorSensor.getColor();
+        return convertColorFromReading(colorMatch.matchClosestColor(reading).color); // This converts no matter what, so it works for specific colors, and doesn't matter for rotations
+    }
+
+    private OutColor convertColorFromReading (Color colorReading) {
+        if (colorReading == redTarget) {
```

Comment:
The color outputs are still jumbled. Detecting red returns GREEN, yellow give RED, etc.